### PR TITLE
Enable combiner lifting for the Group transform

### PR DESCRIPTION
**Pull request by reuvenlax**
R: @slavachernyak 

Note: for Dataflow users, this change will block update. In order to update a use of Group.byFieldNames from before this change, the user must explicitly disable this new behavior by calling withPrecombining(false).

### CODE REVIEW

**Review by slavachernyak — APPROVED** [review]
lgtm

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Group.java [hunk 1]
```diff
@@ -1033,6 +1037,9 @@ static <InputT> CombineFieldsByFields<InputT> of(
           .setSchemaAggregateFn(schemaAggregateFn)
           .setKeyField(keyField)
           .setValueField(valueField)
+          .setFewKeys(
+              true) // We are often selecting only certain fields, so combiner lifting usually
```
Comment (arunpandianp) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Group.java:1041]: will this break update compatibility for jobs already using Group transform?

Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Group.java:1041]: Yes, however there is a workaround (call withPrecombining(false)). IMO we should document this in the Beam release notes along with the workaround.

Comment (arunpandianp) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Group.java:1041]: sgtm

**Comment by reuvenlax** [comment]
run java precommit

**Comment by reuvenlax** [comment]
Run RAT PreCommit

**Comment by reuvenlax** [comment]
Run Java_Examples_Dataflow

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**